provider-service/base/pkg/server/metrics_server.go

```diff
@@ -0,0 +1,79 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sky-uk/kfp-operator/argo/common"
+	"github.com/sky-uk/kfp-operator/provider-service/base/pkg/config"
+)
+
+func InitialiseMetricsServer(ctx context.Context, cfg config.MetricsConfig) (func(), error) {
+	if cfg.Port == 0 {
+		return nil, errors.New("metrics.Port must be specified")
+	}
+
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", cfg.Port))
+	if err != nil {
+		return nil, err
+	}
+
+	return initialiseMetricsServerFromListener(ctx, listener)
+
+}
+
+func initialiseMetricsServerFromListener(ctx context.Context, listener net.Listener) (func(), error) {
+	logger := common.LoggerFromContext(ctx)
+
+	meterProvider, err := newMeterProvider()
+	if err != nil {
+		return nil, err
+	}
+	otel.SetMeterProvider(meterProvider)
+
+	go serveMetrics(ctx, listener)
+
+	return func() {
+		if err := meterProvider.Shutdown(ctx); err != nil {
+			logger.Error(err, "Error shutting down metrics server")
+		}
+	}, nil
+}
+
+func newResource() *resource.Resource {
+	return resource.NewWithAttributes(semconv.SchemaURL,
+		semconv.ServiceName("kfp-operator"),
```

Comment:
```suggestion
		semconv.ServiceName("provider-service"),
```

since is the provider service we are going to be exposing metrics for, perhaps this naming is clearer

as I understand it, this name needs to match whichever name is used in the otel meter to expose custom metrics e.g. response codes, latency etc. which we'll need to add later on in the provider service

Reply:
I would suggest this is driven from the config provided by the specific implementation